### PR TITLE
Fix `az aks get-credentials` on Windows

### PR DESCRIPTION
**Pull request by mboersma**
`NamedTemporaryFile` can't be opened more than once on Windows, so this uses [`mkstemp`](https://docs.python.org/3/library/tempfile.html). Tested interactively on Windows 10, macOS, and Ubuntu Linux.

### CODE REVIEW

**Comment by derekbekoe** [comment]
Should update HISTORY.rst with description of fix.

**Comment by derekbekoe** [comment]
@mboersma There's a conflict after a prev. PR merge.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -1295,13 +1295,17 @@ def aks_get_credentials(client, resource_group_name, name, admin=False,
                 pass
 
         # merge the new kubeconfig into the existing one
-        with tempfile.NamedTemporaryFile(mode='w+t') as additional_file:
+        fd, temp_path = tempfile.mkstemp()
+        additional_file = os.fdopen(fd, 'w+t')
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: `t` is not necessary as `text` mode is the default, but if you prefer explicit, that is fine

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 2]
```diff
@@ -1295,13 +1295,17 @@ def aks_get_credentials(client, resource_group_name, name, admin=False,
...
-            try:
-                merge_kubernetes_configurations(path, additional_file.name)
-            except yaml.YAMLError as ex:
-                logger.warning('Failed to merge credentials to kube config file: %s', ex)
+            merge_kubernetes_configurations(path, temp_path)
+        except yaml.YAMLError as ex:
+            logger.warning('Failed to merge credentials to kube config file: %s', ex)
+        finally:
+            additional_file.close()
+            os.remove(temp_path)
```
Comment (yugangw-msft) [inline src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: I suggest you add a couple of tests for this logic. The related issue impacts quite a few users, so good to build up regression tests

Comment (yugangw-msft) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: Also, a bit straightforward option is to pass the content to `merge_kubernetes_configurations` rather build up a temp file and gets read out again.  Of course this requires some refactor on old code, so up to you.

**Comment by mboersma** [comment]
I'll write some unit tests and rebase this.

**Comment by derekbekoe** [comment]
@mboersma FYI code freeze for next release is 11/8.
cc: @yugangw-msft @troydai

**Comment by troydai** [comment]
@mboersma is this PR still work in progress?

**Comment by mboersma** [comment]
@troydai I want to add tests, but I realized there are already good unit tests around the underlying `_merge_kubernetes_configurations` function. So I added `az aks get-credentials` scenario tests.

But AKS has been so popular we're fighting capacity issues right now, and I've been unable to re-run the scenario test despite trying in the background all day long: `Allocation failed.`

So I'd really like to see this merged because I know it fixes an issue on Windows, but I'm having trouble adding a `get-credentials` test currently. I would be glad to follow on with a PR that adds those tests if that's acceptable to you so we can get this merged, and in the meantime I'll continue to try to get the scenario test to succeed.

**Comment by troydai** [comment]
Okie Dokie. I will merge this PR. I'm interested to know the capacity issue, since we'd like to run automation against live environment routinely as well.

**Comment by BlitzkriegSoftware** [comment]
When will this be shipped?

**Comment by BlitzkriegSoftware** [comment]
When will this be shipped?

**Comment by BlitzkriegSoftware** [comment]
When will this be shipped?

**Comment by troydai** [comment]
I will be released on 11/15.

**Pull request merged**